qcodes/instrument/parameter.py

```diff
@@ -669,8 +669,7 @@ def set_wrapper(value: ParamDataType, **kwargs: Any) -> None:
         return set_wrapper
 
     def get_ramp_values(self, value: Union[Number, Sized],
-                        step: Number = None) -> List[Union[Number,
-                                                           Sized]]:
+                        step: Number = None) -> Sequence[Union[Number, Sized]]:
```

Comment:
Why is this change needed. It seems like all the returned values are lists as far as I can see

Reply:
BTW the type of permissive_range seems to be wrong

Reply:
i don't really remember anymore, but that had to do with typing sweep_values and Sequence being a convariant(?) type as opposed to List. Let me double check this.

> BTW the type of permissive_range seems to be wrong

what should it be? (i'll also double check)

Reply:
For the record. 

List does not work since its invariant in arguments so that you cannot return a List[float] from 
from a function that has List[Union[float, othertype]] 

Sequence is covariant since it's immutable so that is safe. The other alternative would be to explicitly give all the returned variables the full type. 
 

permissive_range is already fixed below